cloudvolume/cloudvolume.py

```diff
@@ -235,14 +272,21 @@ def refresh_provenance(self):
       }
 
     self.provenance = DataLayerProvenance(**provfile)
-    return self
+    self._maybe_cache_provenance()
+    return self.provenance
 
   def commit_provenance(self):
     if self._protocol == 'boss':
-      return self
+      return self.provenance
 
-    self._storage.put_file('provenance', self.provenance.serialize())
-    return self
+    self._storage.put_file('provenance', self.provenance.serialize(), 'application/json')
+    self._maybe_cache_provenance()
+    return self.provenance
+
+  def _maybe_cache_provenance(self):
+    if self.cache and self.provenance:
```

Comment:
i don't think provenance affects functionality, so we should throw a warning if there are differences just in case.

Reply:
Same question as above.